packages/storage/src/platform/browser/connection.ts

```diff
@@ -70,7 +70,7 @@ abstract class XhrConnection<T extends ConnectionType>
     if (this.sent_) {
       throw internalError('cannot .send() more than once');
     }
-    if (isCloudWorkstation(url) && isUsingEmulator) {
+    if (isCloudWorkstation(new URL(url).hostname) && isUsingEmulator) {
```

Comment:
Should we do something like this inside isCloudWorkstation instead, to be defensive about accidentally calling this with more than jsut the hostname?

Reply:
Yeah, I think that's totally fair. Let me go ahead and move this into the util fn